### PR TITLE
Improvements to the Hook contract

### PR DESCRIPTION
**Pull request by nfurfaro**

### CODE REVIEW

File: smart-contracts/contracts/BondingCurveHook.sol [hunk 1]
```diff
@@ -82,6 +82,8 @@ contract BondingCurveHook is ILockKeyPurchaseHookV7 {
   ) external
   {
     require(msg.sender == lockAddress, 'UNAUTHORIZED_ACCESS');
+    address author = bytesToAddress(data);
+    require(msg.sender != author, 'MINT_TO_SELF');
```
Comment (julien51) [inline smart-contracts/contracts/BondingCurveHook.sol]: I think we should remove this. It's so easy to cheat anyway... and in the end, if someone pays the full price of a membership, I guess they can get the share?

**Review by julien51 — APPROVED** [review]

File: smart-contracts/contracts/BondingCurveHook.sol (hunk 1 above)
Comment (nfurfaro) [reply smart-contracts/contracts/BondingCurveHook.sol]: Yes, agreed. This is removed in the most current version (I haven't pushed it yet).

Comment (nfurfaro) [reply smart-contracts/contracts/BondingCurveHook.sol]: fixed.

**Review by julien51 — APPROVED** [review]
I think it's good but would you be able to split this in smaller chunks (linting, dependencies, the buidler stuff, and finally the token stuff)? This way we can more easily review and revert only the stuff that's needed easily!

**Comment by nfurfaro** [comment]
Split into 4 smaller prs, closing.

**Pull request closed without merging**